lib/concurrent/struct/immutable_struct.rb

```diff
@@ -0,0 +1,87 @@
+require 'concurrent/struct/abstract_struct'
+require 'concurrent/synchronization'
+
+module Concurrent
+
+  # A thread-safe, immutable variation of Ruby's standard `Struct`.
+  #
+  # @see http://ruby-doc.org/core-2.2.0/Struct.html Ruby standard library `Struct`
+  module ImmutableStruct
+    include AbstractStruct
+
+    # @!visibility private
+    def initialize(*values)
+      ns_initialize(*values)
+    end
```

Comment:
The problem is that the ImmutableStruct is not safe without `ensure_ivar_visibility!`. Second thread can see the object before its fully initialized. I will update all the child classes with the `Synchronization::Object` changes.